golang/cosmos/x/swingset/keeper/keeper_test.go

```diff
@@ -0,0 +1,158 @@
+package keeper
+
+import (
+	"testing"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/swingset/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func Test_calculateFees(t *testing.T) {
+	type args struct {
+		balances        sdk.Coins
+		submitter, addr sdk.AccAddress
+		powerFlags      []string
+		powerFlagFees   []types.PowerFlagFee
+	}
+	tests := []struct {
+		name   string
+		args   args
+		want   sdk.Coins
+		errMsg string
+	}{
+		{
+			name: "provision pass",
+			args: args{
+				balances:   privilegedProvisioningCoins,
+				submitter:  sdk.AccAddress([]byte("submitter")),
+				addr:       sdk.AccAddress([]byte("addr")),
+				powerFlags: []string{"powerflag1", "powerflag2"},
+			},
+			want: sdk.NewCoins(),
+		},
+		{
+			name: "provision pass and more",
+			args: args{
+				balances:   privilegedProvisioningCoins.Add(privilegedProvisioningCoins...).Add(sdk.NewCoin("coina", sdk.NewInt(1000))),
+				submitter:  sdk.AccAddress([]byte("submitter")),
+				addr:       sdk.AccAddress([]byte("addr")),
+				powerFlags: []string{"powerflag1", "powerflag2"},
+			},
+			want: sdk.NewCoins(),
+		},
+		{
+			name: "cannot pay fee to provision third party",
+			args: args{
+				submitter: sdk.AccAddress([]byte("submitter")),
+				addr:      sdk.AccAddress([]byte("addr")),
+			},
+			errMsg: "submitter is not the same as target address for fee-based provisioning",
```

Comment:
Add power flags for this case so it's not sensitive to the order of validation checks.

Reply:
Done.